tracer/src/Datadog.Trace/Configuration/ImmutableIntegrationSettings.cs

```diff
@@ -0,0 +1,50 @@
+﻿// <copyright file="ImmutableIntegrationSettings.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+namespace Datadog.Trace.Configuration
+{
+    /// <summary>
+    /// Contains integration-specific settings.
+    /// </summary>
+    public class ImmutableIntegrationSettings
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ImmutableIntegrationSettings"/> class from an instance of
+        /// <see cref="IntegrationSettings"/>.
+        /// </summary>
+        /// <param name="settings">The values to use.</param>
+        /// <param name="isExplicitlyDisabled">If true forces the setting Enabled = false. Otherwise, uses <see cref="IntegrationSettings.Enabled"/></param>
+        internal ImmutableIntegrationSettings(IntegrationSettings settings, bool isExplicitlyDisabled)
+        {
+            IntegrationName = settings.IntegrationName;
+            Enabled = isExplicitlyDisabled ? false : settings.Enabled;
+            AnalyticsEnabled = settings.AnalyticsEnabled;
+            AnalyticsSampleRate = settings.AnalyticsSampleRate;
+        }
+
+        /// <summary>
+        /// Gets the name of the integration. Used to retrieve integration-specific settings.
+        /// </summary>
+        public string IntegrationName { get; }
+
+        /// <summary>
+        /// Gets a value indicating whether
+        /// this integration is enabled.
+        /// </summary>
+        public bool? Enabled { get; internal set; }
```

Comment:
`internal set`?

Reply:
Oh, you're right, thanks 🤦‍♂️ I remembered this as how CallTarget was disabling the integration if there was a duck typing error, but that's a completely different approach: https://github.com/DataDog/dd-trace-dotnet/blob/master/tracer/src/Datadog.Trace/ClrProfiler/CallTarget/Handlers/IntegrationOptions.cs#L20

Reply:
Fixed in https://github.com/DataDog/dd-trace-dotnet/pull/1951/commits/25e3e06e35404a2ea0927d8c05ce03a7eb97fafd